airflow/serialization/serde.py

```diff
@@ -260,11 +262,11 @@ def _stringify(classname: str, version: int, value: T | None) -> str:
     s = f"{classname}@version={version}("
     if isinstance(value, _primitives):
         s += f"{value})"
-    elif isinstance(value, _iterables):
-        s += ",".join(str(serialize(value, False)))
+    elif isinstance(value, _collections):
+        s += ",".join(str(serialize(value)))
```

Comment:
`str()` + `join` here looks suspect.  (since `",".join("abc") == "a,b,c"` )

Reply:
A nice addition to the tests along side the round trip tests would be to verify what the serialized form ends up as

Reply:
Good catch. Tests added